packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx

```diff
@@ -0,0 +1,257 @@
+    prefix: string,
+    relatedQuestion: SmartSnippetRelatedQuestion
+  ) {
+    return prefix + (relatedQuestion.expanded ? '-expanded' : '-collapsed');
+  }
+
+  private toggleQuestion(relatedQuestion: SmartSnippetRelatedQuestion) {
+    if (relatedQuestion.expanded) {
+      this.smartSnippetQuestionsList.collapse(relatedQuestion.documentId);
+    } else {
+      this.smartSnippetQuestionsList.expand(relatedQuestion.documentId);
+    }
+  }
+
+  private renderQuestion(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <Button
+        style="text-neutral"
+        part={this.getQuestionPart('question-button', relatedQuestion)}
+        onClick={() => this.toggleQuestion(relatedQuestion)}
+        class="flex items-center px-4"
+        ariaExpanded={`${relatedQuestion.expanded}`}
+        ariaControls={this.getRelatedQuestionId(relatedQuestion)}
+      >
+        <atomic-icon
+          icon={relatedQuestion.expanded ? ArrowDown : ArrowRight}
+          part={this.getQuestionPart('question-icon', relatedQuestion)}
+          class="w-2.5 mr-3 stroke-[1.25]"
+        ></atomic-icon>
+        <Heading
+          level={this.headingLevel ? this.headingLevel + 1 : this.headingLevel}
+          class="text-left text-xl font-bold py-4"
+          part={this.getQuestionPart('question-text', relatedQuestion)}
+        >
+          {relatedQuestion.question}
+        </Heading>
+      </Button>
+    );
+  }
+
+  private renderContent(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <atomic-smart-snippet-answer
+        exportparts="answer"
+        htmlContent={relatedQuestion.answer}
+        innerStyle={this.style}
+      ></atomic-smart-snippet-answer>
+    );
+  }
+
+  private renderSource(relatedQuestion: SmartSnippetRelatedQuestion) {
+    const {source} = relatedQuestion;
+    if (!source) {
+      return [];
+    }
+    const interactiveResult = buildInteractiveResult(this.bindings.engine!, {
+      options: {result: source},
+    });
+    return (
+      <footer
+        part="footer"
+        aria-label={this.bindings.i18n.t('smart-snippet-source')}
+      >
+        <LinkWithResultAnalytics
+          title={source.clickUri}
+          href={source.clickUri}
+          target="_self"
+          part="source-url"
+          onSelect={() => interactiveResult.select()}
+          onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+          onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+        >
+          {source.clickUri}
+        </LinkWithResultAnalytics>
+        <LinkWithResultAnalytics
+          title={source.title}
+          href={source.clickUri}
+          target="_self"
+          part="source-title"
+          onSelect={() => interactiveResult.select()}
+          onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+          onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+        >
+          {source.title}
```

Comment:
Can you add a todo for Simo here? https://coveord.atlassian.net/browse/KIT-1523

Reply:
Maybe the footer can be extracted to a functional component as well 🤔

Reply:
Updated the doc of the JIRA instead